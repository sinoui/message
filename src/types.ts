/**
 * 消息配置
 */
export interface MessageConfig {
  /**
   * 消息显示时长
   */
  duration?: number;
  /**
   * 消息主键
   */
  key?: string;
}

/**
 * 整个消息模块的配置
 */
export interface Settings {
  /**
   * 消息显示时长
   */
  duration?: number;
  /**
   * 消息显示最大个数
   */
  max?: number;
}

export type MessageType = 'info' | 'loading' | 'warning' | 'success' | 'error';

/**
 * 消息接口
 */
export interface MessageInterface {
  /**
   * 消息类型
   *
   * @type {('info' | 'loading' | 'warning' | 'success' | 'error')}
   * @memberof MessageInterface
   */
  type: MessageType;
  /**
   * 消息显示时长
   *
   * @type {number}
   * @memberof MessageInterface
   */
  duration: number;
  /**
   * 消息内容
   *
   * @type {React.ReactNode}
   */
  content: React.ReactNode;
  /**
   * 消息主键
   */
  key: string;
}
