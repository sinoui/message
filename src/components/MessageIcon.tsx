import React from 'react';
import Icon from 'sinoui-components/Icon';
import { MdInfo, MdWarning, MdCheckCircle, MdError } from 'react-icons/md';
import styled from 'styled-components';
import Progress from 'sinoui-components/Progress';
import { MessageType } from '../types';

const DenseIcon = styled(Icon)`
  font-size: 16px;
  margin-right: 8px;
  display: inline-flex;
  height: 20px;
  align-items: center;
`;

const DenseProgress = styled(Progress)`
  margin-right: 8px;
  margin-top: 3px;
`;

const renderIcon = (type: MessageType) => {
  switch (type) {
    case MessageType.success:
      return <MdCheckCircle />;
    case MessageType.error:
      return <MdError />;
    case MessageType.warning:
      return <MdWarning />;
    default:
      return <MdInfo />;
  }
};

function MessageIcon({ type }: { type: MessageType }) {
  return type === MessageType.loading ? (
    <DenseProgress size={16} thickness={1.5} />
  ) : (
    <DenseIcon color={type}>{renderIcon(type)}</DenseIcon>
  );
}

export default React.memo(MessageIcon);
