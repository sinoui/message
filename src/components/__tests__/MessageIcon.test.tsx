import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import TestWrapper from './TestWrapper';
import MessageIcon from '../MessageIcon';
import { MessageType } from '../../types';

describe('MessageIcon 镜像测试', () => {
  it('info类型', () => {
    const tree = renderer
      .create(
        <TestWrapper>
          <MessageIcon type={MessageType.info} />
        </TestWrapper>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('warning类型', () => {
    const tree = renderer
      .create(
        <TestWrapper>
          <MessageIcon type={MessageType.warning} />
        </TestWrapper>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('error类型', () => {
    const tree = renderer
      .create(
        <TestWrapper>
          <MessageIcon type={MessageType.error} />
        </TestWrapper>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('success类型', () => {
    const tree = renderer
      .create(
        <TestWrapper>
          <MessageIcon type={MessageType.success} />
        </TestWrapper>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('loading类型', () => {
    const tree = renderer
      .create(
        <TestWrapper>
          <MessageIcon type={MessageType.loading} />
        </TestWrapper>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
