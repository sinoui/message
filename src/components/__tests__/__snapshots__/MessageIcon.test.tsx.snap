// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageIcon 镜像测试 error类型 1`] = `
.c0 {
  font-family: 'Material Icons';
  font-weight: normal;
  font-style: normal;
  font-size: 24px;
  max-width: 1em;
  display: inline-block;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  word-wrap: normal;
  white-space: nowrap;
  direction: ltr;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  -moz-osx-font-smoothing: grayscale;
  font-feature-settings: 'liga';
  color: currentColor;
  -webkit-transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  color: #e53935;
  font-size: 16px;
  margin-right: 8px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 20px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<span
  className="c0 sinoui-icon"
  color="error"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
    />
  </svg>
</span>
`;

exports[`MessageIcon 镜像测试 info类型 1`] = `
.c0 {
  font-family: 'Material Icons';
  font-weight: normal;
  font-style: normal;
  font-size: 24px;
  max-width: 1em;
  display: inline-block;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  word-wrap: normal;
  white-space: nowrap;
  direction: ltr;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  -moz-osx-font-smoothing: grayscale;
  font-feature-settings: 'liga';
  color: currentColor;
  -webkit-transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  color: #3f51b5;
  font-size: 16px;
  margin-right: 8px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 20px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<span
  className="c0 sinoui-icon"
  color="primary"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
    />
  </svg>
</span>
`;

exports[`MessageIcon 镜像测试 loading类型 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 16px;
  height: 16px;
  -webkit-animation: kOBPdp 1.7s linear infinite;
  animation: kOBPdp 1.7s linear infinite;
}

.c1 {
  margin-right: 8px;
  margin-top: 3px;
}

<div
  className="c0 sinoui-progress c1"
  color="primary"
  size={16}
>
  <canvas />
</div>
`;

exports[`MessageIcon 镜像测试 success类型 1`] = `
.c0 {
  font-family: 'Material Icons';
  font-weight: normal;
  font-style: normal;
  font-size: 24px;
  max-width: 1em;
  display: inline-block;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  word-wrap: normal;
  white-space: nowrap;
  direction: ltr;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  -moz-osx-font-smoothing: grayscale;
  font-feature-settings: 'liga';
  color: currentColor;
  -webkit-transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  color: #00c853;
  font-size: 16px;
  margin-right: 8px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 20px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<span
  className="c0 sinoui-icon"
  color="success"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
    />
  </svg>
</span>
`;

exports[`MessageIcon 镜像测试 warning类型 1`] = `
.c0 {
  font-family: 'Material Icons';
  font-weight: normal;
  font-style: normal;
  font-size: 24px;
  max-width: 1em;
  display: inline-block;
  text-transform: none;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  word-wrap: normal;
  white-space: nowrap;
  direction: ltr;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  -moz-osx-font-smoothing: grayscale;
  font-feature-settings: 'liga';
  color: currentColor;
  -webkit-transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  transition: color 200ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  color: #ffc107;
  font-size: 16px;
  margin-right: 8px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 20px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<span
  className="c0 sinoui-icon"
  color="warning"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
    />
  </svg>
</span>
`;
